### PR TITLE
feat(preview) : #MAG-505 add file preview

### PR DESCRIPTION
**Pull request by FlorentMr**
## Describe your changes
Add preview of multiple files such as PDF, word documents, ...

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: frontend/src/components/card-content-file/CardContentFile.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  };
+
+  return (
+    <>
+      <FileInfos
+        fileName={card.metadata.filename}
+        owner={cardDocument?.ownerName ?? ""}
+        size={size.value + size.unit}
+        fileType={extensionText}
+        onDownload={() => console.log("Télécharger le fichier")}
```
Comment (ThomasSevagen) [inline frontend/src/components/card-content-file/CardContentFile.tsx:34]: tej les logs et les méthodes si elles servent qu'à log

File: frontend/src/components/file-infos/FileInfos.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  );
+
+  const isHorizontal =
+    isPrimaryBreakpoint || (isCommentPanelOpen && isSecondaryBreakpoint);
+
+  return (
+    <>
+      <h1
+        style={{
+          paddingTop: "2rem",
```
Comment (ThomasSevagen) [inline frontend/src/components/file-infos/FileInfos.tsx:68]: export

File: frontend/src/components/card-content-file/CardContentFile.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import { useBoard } from "~/providers/BoardProvider";
+
+export const CardContentFile: FC<CardContentFileProps> = ({ card }) => {
+  const { documents, displayModals } = useBoard();
+  const cardDocument = documents.find((doc) => doc._id === card.resourceId);
+  const extensionText = useFileExtensionDescription(card.metadata.extension);
+  const size = useFileSize(card.metadata.size);
+
+  const isOfficePdf = () => {
+    const ext = ["doc", "ppt", "odt"];
```
Comment (ThomasSevagen) [inline frontend/src/components/card-content-file/CardContentFile.tsx:18]: enum

File: frontend/src/components/card-content-file/CardContentFile.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    return ext.includes(card.metadata.extension);
+  };
+
+  const isOfficeExcelOrCsv = () => {
+    const ext = ["xls", "csv", "xlsx"];
```
Comment (ThomasSevagen) [inline frontend/src/components/card-content-file/CardContentFile.tsx:23]: enum

File: frontend/src/components/preview-content/utils.tsx [hunk 5]
```diff
@@ -13,6 +13,7 @@ import { Card } from "~/models/card.model";
 
 export const displayPreviewContentByType = (card: Card) => {
   const cardType = card.resourceType as RESOURCE_TYPE;
+  console.log(card);
```
Comment (ThomasSevagen) [inline frontend/src/components/preview-content/utils.tsx:16]: log

File: frontend/src/core/enums/file-extension.enum.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
+export enum FILE_EXTENSION {
```
Comment (ThomasSevagen) [inline frontend/src/core/enums/file-extension.enum.ts:1]: ducoup tu peux utiliser cet enum pour remplir tes variables plus haut
cf comms enums plus haut

File: frontend/src/hooks/useFileSize.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import { useMemo } from "react";
+
+type FileSizeUnit = "B" | "KB" | "MB" | "GB" | "TB";
```
Comment (ThomasSevagen) [inline frontend/src/hooks/useFileSize.ts:3]: enum

File: frontend/src/providers/BoardProvider/index.tsx [hunk 8]
```diff
@@ -70,6 +72,11 @@ export const BoardProvider: FC<BoardProviderProps> = ({ children }) => {
       : new Board();
   }, [boardData]);
 
+  const documents = useMemo(() => {
+    console.log(documentsData);
```
Comment (ThomasSevagen) [inline frontend/src/providers/BoardProvider/index.tsx:76]: log

File: frontend/src/theme.d.ts [hunk 9]
```diff
@@ -0,0 +1,10 @@
+declare module "@mui/material/styles" {
+  interface BreakpointOverrides {
+    xs: true;
```
Comment (ThomasSevagen) [inline frontend/src/theme.d.ts:3]: [EXT_ENUM_XS]: true etc

File: frontend/vite.config.ts [hunk 10]
```diff
@@ -14,26 +14,26 @@ export default ({ mode }: { mode: string }) => {
   // Proxy variables
   const headers = hasEnvFile
     ? {
-      "set-cookie": [
-        `oneSessionId=${envs.VITE_ONE_SESSION_ID}`,
-        `XSRF-TOKEN=${envs.VITE_XSRF_TOKEN}`,
-      ],
-      "Cache-Control": "public, max-age=300",
-    }
+        "set-cookie": [
```
Comment (ThomasSevagen) [inline frontend/vite.config.ts:17]: à quoi ça sert?

File: frontend/vite.config.ts [hunk 11]
```diff
@@ -82,6 +82,10 @@ export default ({ mode }: { mode: string }) => {
     headers,
     open: true,
     strictPort: true,
+    mimeTypes: {
```
Comment (ThomasSevagen) [inline frontend/vite.config.ts:85]: à quoi ça sert?

File: frontend/src/hooks/useFileExtensionDescription.ts
Comment (niko-amosse) [inline frontend/src/hooks/useFileExtensionDescription.ts:1]: pas besoin d'i18n ?

File: frontend/src/hooks/useFileSize.ts
Comment (niko-amosse) [inline frontend/src/hooks/useFileSize.ts:1]: https://www.npmjs.com/package/filesize

**Review by niko-amosse — APPROVED** [review]

**Review by alicedraillard — APPROVED** [review]

**Pull request merged**